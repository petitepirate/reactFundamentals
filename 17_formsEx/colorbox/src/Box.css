.Box {
	margin: 10px auto;
}
