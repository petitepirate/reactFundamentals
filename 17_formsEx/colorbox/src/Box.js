import React from 'react';
import './Box.css'

function Box({idx, removeBox, width, height, boxBackgroundColor}) {
  return (
    <div>
      <div 
      className='Box'
      key={idx}
      style={{
        width: `${width}px`,
        height: `${height}px`,
        backgroundColor: `${boxBackgroundColor}`
      }}>
        <button type="button" onClick={() => removeBox(idx)}>Remove X</button>
      </div>
    </div>
  )
}

export default Box;
