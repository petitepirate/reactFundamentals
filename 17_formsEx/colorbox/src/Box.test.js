import React from 'react';
import { render } from '@testing-library/react';
import Box from './Box'

it('renders Box component without crashing', () => {
  render(<Box />)
})

it('matches snapshot', () => {
  const { asFragment } = render(<Box 
    idx='1'
    width='200'
    height='200'
    boxBackgroundColor='teal' />)
  expect(asFragment()).toMatchSnapshot();
})
