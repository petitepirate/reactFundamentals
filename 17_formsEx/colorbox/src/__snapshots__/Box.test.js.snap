// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="Box"
      style="width: 200px; height: 200px; background-color: teal;"
    >
      <button
        type="button"
      >
        Remove X
      </button>
    </div>
  </div>
</DocumentFragment>
`;
